### ISSUE TITLE
Inconsistent `onEnter` and `onLeave` with `SET`

### ISSUE THREAD

**Issue by 96LawDawg** [ArnoldSmith86/virtualtabletop#1836]
Assume a room with holder1 and holder2,  holder1 has `onEnter` and `onLeave` properties established.  When you drag and drop things in holder1, the `onEnter` and `onLeave` work properly.  When you progamatically use `MOVE` to move back and forth between the holders, everything works.

But, when you use `SET` parent to or from holder1, that is where things break down.  `SET` parent to holder1 triggers the `onEnter`, but when you take a widget that is in holder1 and `SET` its parent to holder2, `onLeave` does not get triggered.

This is sort of related to #1212, but in that issue, there is no discussion of `SET` parents.

**Comment by robartsd** [comment]
My personal preference would be to have SET parent not trigger onEnter or onLeave, allowing SET to be used a a low level way to make changes without side effects.

**Comment by mousewax** [comment]
How do I programmatically move a card to a holder that is not already in another holder and have onEnter get applied (since MOVE is only from and to holders)? Would enterRoutines still get fired? I would rather have a consistent behavior among all of these. Why would I use automation to put a card in a holder and not want what I specifically put on that holder to take effect? I think that when a card enters a holder by whatever means, the onEnter needs to fire.

**Comment by robartsd** [comment]
Not being able to MOVE a collection is the root problem there - if we make MOVE work with collections, you would be able to MOVE the card directly to the holder or SET it's parent depending on if you want the holder side effects to take place or not. For now you could work around by using SET to place in a hidden holder then MOVE to put in the target holder.

**Comment by 96LawDawg** [comment]
I agree with @mousewax.  If I don't want a routine to fire for some reason, I'll turn it off.  But dragging/dropping, MOVE, and SET should all trigger holder onEnter and enterRoutine and the opposite when they leave.

**Comment by 96LawDawg** [comment]
Same issue reported in #109
